src/osa/utils/utils.py

```diff
@@ -285,3 +292,137 @@ def wait_for_daytime(start=8, end=18):
+    pattern = r'^dec_(\d{3,4})$|^dec_min_(\d{3,4})$'
+    if re.match(pattern, dec_str):
+        
+        # Split the string into parts
+        parts = dec_str.split('_')
+
+        # Extract the sign, degrees, and minutes
+        sign = 1 if 'min' not in parts else -1
+        degrees = int(parts[-1])
+
+        # Calculate the numerical value
+        dec_value = sign * (degrees / 100)
+
+        return dec_value*u.deg
+
+
+def get_declinations_dict(list1: list, list2: list) -> dict:
+    """Return a dictionary created from two given lists."""
+    corresponding_dict = {}
+    for index, element in enumerate(list2):
+        corresponding_dict[element] = list1[index]
+    return corresponding_dict
+        
+
+def get_nsb_dict(rf_models_dir: Path, rf_models_prefix: str) -> dict:
+    """Return a dictionary with the NSB level of the RF models and the path to each model."""
+    rf_models = sorted(rf_models_dir.glob(f"{rf_models_prefix}*"))
+    pattern = r"nsb_tuning_([\d.]+)"
+    nsb_dict = {
+        float(re.search(pattern, str(rf_model)).group(1)): rf_model
+        for rf_model in rf_models if re.search(pattern, str(rf_model))
+    }
+    return nsb_dict
+
+
+def get_mc_nsb_dir(run_id: int, rf_models_dir: Path) -> Path:
+    """
+    Return the path of the RF models directory with the NSB level 
+    closest to that of the data for a given run.
+    """
+    input_dir = Path(options.directory)
+    _, additional_nsb, _ = find_tailcuts(input_dir, run_id)
+
+    rf_models_prefix = cfg.get("lstchain", "mc_prod")
+    nsb_dict = get_nsb_dict(rf_models_dir, rf_models_prefix)
+    closest_nsb_value = min(nsb_dict.keys(), key=lambda x: abs(float(x) - additional_nsb))
+
+    return nsb_dict[closest_nsb_value]
+
+
+def get_RF_model(run_id: int) -> Path:
+    """Get the path of the RF model to be used in the DL2 production for a given run."""
+    datacheck_dir = options.directory / options.dl1_prod_id / "datacheck"
+    datacheck_file = datacheck_dir / f"datacheck_dl1_LST-1.Run{run_id:05d}.h5"
+
+    source_dec = get_median_dec(datacheck_file)
+    source_culmination = culmination_angle(source_dec)
+
+    rf_models_base_dir = Path(cfg.get("LST1", "RF_MODELS"))
+    rf_models_dir = get_mc_nsb_dir(run_id, rf_models_base_dir)
+    
+    if options.test:
+        rf_model_path = rf_models_dir / "dec_2276"
+        return rf_model_path.resolve()
+  
+    dec_list = os.listdir(rf_models_dir)
+
+    # Convert each string in the list to numerical values
+    dec_values = [convert_dec_string(dec) for dec in dec_list]
+    dec_values = [dec for dec in dec_values if dec is not None]
+    
+    closest_declination = min(dec_values, key=lambda x: abs(x - source_dec))
+    closest_dec_culmination = culmination_angle(closest_declination)
+    
+    location = observatory_locations["cta_north"]
+    Lat = location.lat  # latitude of the LST1 site    
+    closest_lines = sorted(sorted(dec_values, key=lambda x: abs(x - Lat))[:2])
```

Comment:
```suggestion
    lst_location = observatory_locations["cta_north"]
    lst_latitude = lst_location.lat  # latitude of the LST1 site    
    closest_lines = sorted(sorted(dec_values, key=lambda x: abs(x - lst_latitude))[:2])
```
more explicit names for ease the reading of the code